src/warpychain/client.go

```diff
@@ -0,0 +1,146 @@
+		Raw(`SELECT finished_block_height
+		FROM sync_state
+		WHERE name = 'WarpyChainer';`).
+		Scan(&lastSyncedHeight).Error
+
+	return
+}
+
+func (self *WarpyChainClient) getCurrentBlockHeight() (currentBlockHeight int64, currentBlockHash string, err error) {
+	header, err := self.ethClient.HeaderByNumber(context.Background(), nil)
+	if err != nil {
+		self.Log.WithError(err).Error("Could not get ETH header")
+		return
+	}
+
+	currentBlockHeight = header.Number.Int64()
+	currentBlockHash = header.TxHash.String()
+
+	self.Log.WithField("height", currentBlockHeight).WithField("hash", currentBlockHash).Debug("Current block height")
+
+	return
+}
+
+func (self *WarpyChainClient) getBlockInfo(blockNumber int64) (blockInfo *types.Block, err error) {
+	blockInfo, err = self.ethClient.BlockByNumber(context.Background(), big.NewInt(blockNumber))
+	if err != nil {
+		self.Log.WithError(err).Error("Could not get ETH header")
+	}
+	return
+}
+
+func (self *WarpyChainClient) updateSyncState(ctx context.Context, blockHeight int64, blockHash string) (err error) {
+	err = self.db.WithContext(ctx).
+		Exec(`UPDATE sync_state
+	SET finished_block_height = ?, finished_block_hash = ?
+	WHERE name = 'WarpyChainer';`, blockHeight, blockHash).
+		Error
+
+	if err != nil {
+		self.Log.WithError(err).Error("Could not update sync_state table")
+		return
+	}
+	return
+}
+
+func (self *WarpyChainClient) checkForRedStoneTx(tx *types.Transaction, ctx context.Context) (isRedStoneTx bool) {
+	isRedStoneTx = false
+	if tx.Hash().Hex() == "0x04659c685df5e87d9bb3755d3943442c25bdf773113fa54cf5427aa38729235e" {
+		encodedString := hex.EncodeToString(tx.Data())
+		if strings.Contains(encodedString, "000002ed57011e0000") {
+			fmt.Println("RedStone Transaction!")
+			isRedStoneTx = true
+		}
+	}
+	return
+}
+
+func (self *WarpyChainClient) getTxSenderHash(tx *types.Transaction) (txSenderHash string, err error) {
+	signer := types.LatestSignerForChainID(tx.ChainId())
+	sender, err := types.Sender(signer, tx)
+	txSenderHash = sender.Hash().String()
+	return
+}
+
+func (self *WarpyChainClient) writeInteractionToWarpy(ctx context.Context, tx *types.Transaction, arweaveSigner string) (interactionId string, err error) {
+	self.Log.WithField("txId", tx.Hash()).Debug("Writing interaction to Warpy...")
+	signer, err := bundlr.NewArweaveSigner(arweaveSigner)
+	if err != nil {
+		self.Log.WithError(err).Warn("Could not create Arweave Signer")
+		return
+	}
+	sender, err := self.getTxSenderHash(tx)
+	if err != nil {
+		self.Log.WithError(err).WithField("txId", tx.Hash()).Warn("Could not retrieve tx sender")
+		return
+	}
+	input := WarpyChainInput{
+		Function: "addPointsCsv",
+		Points:   20,
+		AdminId:  "769844280767807520",
+		Members:  []Member{{Id: sender, Roles: []string{}}},
+		NoBoost:  true,
+	}
+
+	interactionId, err = self.writeInteractionClient.WriteInteraction(ctx, input, sequencer_types.WriteInteractionOptions{ContractTxId: "JkWVDtfoCNt1-RmbhcttpQBV05o6MX1qB3k3DiXCFIE"}, signer)
```

Comment:
hmm, this does not look like a Warpy contract's tx id...at least not the 'mainnet' one. Please move this id to env. props.